ext/nnstreamer/tensor_filter/tensor_filter_onnxruntime.cc

```diff
@@ -0,0 +1,564 @@
+
+    // Get output shapes/dims
+    size_t num_dims
+        = session.GetOutputTypeInfo (i).GetTensorTypeAndShapeInfo ().GetDimensionsCount ();
+    output_shapes[i]
+        = session.GetOutputTypeInfo (i).GetTensorTypeAndShapeInfo ().GetShape ();
+
+    // free dimensions are treated as 1 if not overriden
+    for (int64_t &dim : output_shapes[i]) {
+      if (dim == -1) {
+        dim = 1;
+      }
+    }
+
+    if (getTensorDim (outputInfo.info[i].dimension, output_shapes[i], num_dims)) {
+      cleanup ();
+      throw std::invalid_argument ("Shape output rank too high.");
+    }
+    outputInfo.info[i].name = nullptr;
+
+    gchar *dim;
+    dim = gst_tensor_get_dimension_string (outputInfo.info[i].dimension);
+
+    nns_logd ("outputInfo[%zu] >> name[%s], type[%d], dim[%s]", i,
+        outputInfo.info[i].name, outputInfo.info[i].type, dim);
+    g_free (dim);
+  }
+
+  empty_model = false;
+  allocator = Ort::AllocatorWithDefaultOptions{ nullptr }; /* delete unique_ptr */
+}
+
+/**
+ * @brief Method to execute the model.
+ */
+void
+onnxruntime_subplugin::invoke (const GstTensorMemory *input, GstTensorMemory *output)
+{
+  size_t i;
+  assert (!empty_model);
+
+  input_tensors.clear ();
+  output_tensors.clear ();
+
+  memInfo = Ort::MemoryInfo::CreateCpu (
+      OrtAllocatorType::OrtArenaAllocator, OrtMemType::OrtMemTypeDefault);
+
+  assert (input_shapes.size () == inputInfo.num_tensors);
+
+  if (!input)
+    throw std::runtime_error ("Invalid input buffer, it is NULL.");
+  if (!output)
+    throw std::runtime_error ("Invalid output buffer, it is NULL.");
+
+  // Set input to tensor
+  for (i = 0; i < inputInfo.num_tensors; ++i) {
+    input_tensors.emplace_back (Ort::Value::CreateTensor (memInfo, input[i].data,
+        input[i].size, input_shapes[i].data (), input_shapes[i].size (), input_types[i]));
+  }
+
+  // double-check the dimensions of the input tensor
+  assert (input_tensors[0].IsTensor ()
+          && input_tensors[0].GetTensorTypeAndShapeInfo ().GetShape () == input_shapes[0]);
+
+  // Set output to tensor
+  for (i = 0; i < outputInfo.num_tensors; ++i) {
+    output_tensors.emplace_back (
+        Ort::Value::CreateTensor (memInfo, output[i].data, output[i].size,
+            output_shapes[i].data (), output_shapes[i].size (), output_types[i]));
+  }
+
+  try {
+    // call Run() to fill in the GstTensorMemory *output data with the probabilities of each
+    session.Run (Ort::RunOptions{ nullptr }, input_names.data (),
+        input_tensors.data (), input_names.size (), output_names.data (),
+        output_tensors.data (), output_names.size ());
+
+    // double-check the dimensions of the output tensor
+    assert (output_tensors.size () == output_names.size () && output_tensors[0].IsTensor ());
+
+  } catch (const Ort::Exception &exception) {
+    std::cerr << "ERROR running model inference: " << exception.what () << std::endl;
+    cleanup ();
+    throw std::invalid_argument ("exception.");
```

Comment:
Check the throw error type. Is this really an "invalid argument"?

Reply:
I modified the code to **std::runtime_error**